src/journal-remote/journal-gatewayd.c

```diff
@@ -181,11 +183,12 @@ static ssize_t request_reader_entries(
                 } else if (r == 0) {
 
                         if (m->follow) {
-                                r = sd_journal_wait(m->journal, (uint64_t) -1);
+                                r = sd_journal_wait(m->journal, (uint64_t) JOURNAL_WAIT_TIMEOUT);
                                 if (r < 0) {
                                         log_error_errno(r, "Couldn't wait for journal event: %m");
                                         return MHD_CONTENT_READER_END_WITH_ERROR;
-                                }
+                                } else if (r == SD_JOURNAL_NOP)
```

Comment:
If we don't have the if here we will hit the continue and restart the loop and not return and the results would be the same as if we hadn't added the timeout. This needs to break out of the loop so that this function can return to microhttpd so that it can check on the socket.